### DISCUSSION TITLE
can I use gcc and clang linter differently?

### DISCUSSION THREAD

**Discussion by laoshaw** [w0rp/ale#4156] [Q&A - Ask for help with problems]
when I work with c or c++ file I noticed ALEInfo shows "Linter Aliases: 'cc' -> ['gcc', 'clang']", and "Enabled Linters:['cc']", how can I use just one of them(be it gcc or clang) but not both? is it do-able? I tried to set :  "let g:ale_linters={'c': ['gcc']}" but it always insists on "cc" which consists of gcc and clang.
The reason is that I use gcc, and some of its flags(e.g. -fanalyzer) is not supported by clang.

**Comment by hsanson** [comment]
From the help it seems that setting `g:ale_cpp_cc_executable` to `gcc` would make ALE use gcc instead of clang:

```
g:ale_cpp_cc_executable                               *g:ale_cpp_cc_executable*
                                                      *b:ale_cpp_cc_executable*
  Type: |String|
  Default: `'<auto>'`

  This variable can be changed to use a different executable for a C++ compiler.

  ALE will try to use `clang++` if Clang is available, otherwise ALE will
  default to checking C++ code with `gcc`.

```

**Reply by laoshaw** [reply]
that does not really help, the "cc"(clang and gcc, or clang++ and g++) will still be there. there is no way I can use just one of each, they always go together in pair no matter what: always "Enabled Linters:['cc']" instead of "Enabled Linters:['g++']" or "Enabled Linters:['clang']" etc
